include/veriblock/blockchain/commands/addendorsement.hpp

```diff
@@ -48,19 +48,13 @@ struct AddEndorsement : public Command {
     auto minHeight = (std::max)(containing->height - window, 0);
     Chain<protected_index_t> chain(minHeight, containing);
 
-    auto endorsedHeight = e_->endorsedHeight;
-    if (containing->height - endorsedHeight > window) {
-      return state.Invalid(protected_block_t::name() + "-expired",
-                           "Endorsement expired");
-    }
-
-    auto* endorsed = chain[endorsedHeight];
+    auto* endorsed = ed_->getBlockIndex(e_->endorsedHash);
     if (!endorsed) {
       return state.Invalid(protected_block_t::name() + "-no-endorsed-block",
-                           "No block found on endorsed block height");
+                           "No block found the tree");
     }
 
-    if (endorsed->getHash() != e_->endorsedHash) {
+    if (endorsed->getHash() != chain[endorsed->height]->getHash()) {
```

Comment:
Need additional check that chain[endorsed->height] exists. It seems you've just removed this check.

Reply:
I have fixed it